Src/WindowsServer/WindowsServer.Shared.Tests/AzureRoleEnvironmentTelemetryInitializerTest.cs

```diff
@@ -69,33 +47,83 @@ public void AzureRoleEnvironmentTelemetryInitializerDoesNotOverrideRoleInstance(
         {
-            var telemetryItem = new EventTelemetry();
+            var telemetryItem = new EventTelemetry();            
             AzureRoleEnvironmentTelemetryInitializer initializer = new AzureRoleEnvironmentTelemetryInitializer();
-            AzureRoleEnvironmentContextReader.BaseDirectory = ServiceRuntimeHelper.TestWithServiceRuntimePath;
-            AzureRoleEnvironmentContextReader.Instance = null;
-            ServiceRuntimeHelper.IsAvailable = true;
-
             telemetryItem.Context.GetInternalContext().NodeName = "Test";
             initializer.Initialize(telemetryItem);
 
             Assert.Equal("Test", telemetryItem.Context.GetInternalContext().NodeName);
         }
 
         [TestMethod]
-        public void AzureRoleEnvironmentTelemetryInitializerSetsTelemetryContextPropertiesWhenRoleEnvironmentIsNotAvailable()
-        {
-            var telemetryItem = new EventTelemetry();
+        public void AzureRoleEnvironmentTelemetryInitializerSetsTelemetryContextPropertiesToNullWhenNotRunningInsideAzureCloudService()
+        {            
+            // This test asssumes that it is not running inside a cloud service.
+            // Its Ok even if Azure ServiceRunTime dlls are in the GAC, as IsAvailable() will return false, and hence 
+            // no context will be further attempted to be read.
+            var telemetryItem = new EventTelemetry();            
             AzureRoleEnvironmentTelemetryInitializer initializer = new AzureRoleEnvironmentTelemetryInitializer();
-            AzureRoleEnvironmentContextReader.BaseDirectory = ServiceRuntimeHelper.TestWithServiceRuntimePath;
-            AzureRoleEnvironmentContextReader.Instance = null;
-
-            ServiceRuntimeHelper.IsAvailable = false;
-            initializer.Initialize(telemetryItem);
-            ServiceRuntimeHelper.IsAvailable = true;
+            initializer.Initialize(telemetryItem);            
 
             Assert.Null(telemetryItem.Context.Cloud.RoleName);
             Assert.Null(telemetryItem.Context.Cloud.RoleInstance);
             Assert.Null(telemetryItem.Context.GetInternalContext().NodeName);
         }
+
+        [TestMethod]
+        [Description("Validates that requested DLL was loaded into separate AppDomain and not to the current domain.")]
+        public void AzureRoleEnvironmentTelemetryInitializerLoadDllToSeparateAppDomain()
+        {
+            // A random dll which is not already loaded to the current AppDomain but dropped into bin folder.
+            string dllPath = Path.Combine(AppDomain.CurrentDomain.SetupInformation.ApplicationBase, "Newtonsoft.Json.dll");
+                        
+            try
+            {
+                // Publish the dll to GAC to give a chance for  AzureRoleEnvironmentTelemetryInitializer to load it to a new AppDomain
+                new System.EnterpriseServices.Internal.Publish().GacInstall(dllPath);
```

Comment:
Mark test as "Inconclusive" if failed to add to GAC. I never run this solutions "As Administrator"